### PR TITLE
push: remove hyper and add a naive http client

### PR DESCRIPTION
**Pull request by overvenus**
PTAL @siddontang @BusyJay

### CODE REVIEW

File: src/push.rs [hunk 1]
```diff
@@ -132,23 +126,61 @@ fn push(job: &str,
...
+
+        let mut bw = BufWriter::new(stream.by_ref());
+        bw.write_fmt(format_args!("{} {} HTTP/1.1\r\n", method, push_url.path()))?;
+        bw.write_fmt(format_args!("Host: {}\r\n", push_url.host_str().unwrap()))?;
+        bw.write_fmt(format_args!("Content-Type: {}\r\n", encoder.format_type()))?;
+        bw.write_fmt(format_args!("Content-Length: {}\r\n", buf.len()))?;
+        bw.write_all(b"Connection: Keep-Alive\r\n")?;
+        bw.write_all(b"\r\n")?;
+        bw.write_all(&buf)?;
+        bw.flush()?;
```
Comment (siddontang) [inline src/push.rs]: if send error, not remove the stream from the map?

**Comment by siddontang** [comment]
@BusyJay, It seems not easier than using hyper directly.

**Comment by ngaut** [comment]
There is no need to do something like that.  Are you going to write http protocol again and again when you need to use http protocol? Come on, that's why we need library. That's why those great guys create hyper. Please don't trying to hand write http protocol or something similar. Re-use crate instead.

**Comment by BusyJay** [comment]
I don't think it's always suitable to re-use crate. It's also important for a library to minimize its dependencies. So if some functionality is very easy to implement, I will prefer implementing it instead of introducing another dependency. The HTTP implementation here looks complicated than we thought, so I agree to reuse hyper or something more lightweight.

**Pull request closed without merging**